drivers/i2c/i2c_sam0.c

```diff
@@ -480,6 +484,8 @@ static int i2c_sam0_transfer(const struct device *dev, struct i2c_msg *msgs,
 			/* return the bus to idle */
 			i2c->CTRLB.bit.CMD = 3;
 
+			k_sem_give(&data->lock);
+
```

Comment:
Could this result in multiple "give" ?

Reply:
This relates to the single exit path from the function being preferred, lock on the way in, unlock on the single path out